srtcore/utilities.h

```diff
@@ -843,6 +843,126 @@ struct CallbackHolder
+    //  - 50, 51, 100, 55, 80, 1000, 600, 1500, 1200, 10, 90
+    // 2. This window is now sorted, but we only know the value in the middle:
+    //  - 10, 50, 51, 55, 80, [[90]], 100, 600, 1000, 1200, 1500
+    // 3. Now calculate:
+    //   - lower: 90/8 = 11.25
+    //   - upper: 90*8 = 720
+    // 4. Now calculate the arithmetic median from all these values,
+    //    but drop those from outside the <lower, upper> range:
+    //  - 10, (11<) [ 50, 51, 55, 80, 90, 100, 600, ] (>720) 1000, 1200, 1500
+    // 5. Calculate the median from the extracted range,
+    //    NOTE: the median is actually repeated once, so size is +1.
+    //
+    //    values = { 50, 51, 55, 80, 90, 100, 600 };
+    //    sum = 90 + accumulate(values); ==> 1026
+    //    median = sum/(1 + values.size()); ==> 147
+    //
+    // For comparison: the overall arithmetic median from this window == 430
+    //
+    // 6. Returned value = 1M/median
+
+    // get median value, but cannot change the original value order in the window
+    std::copy(window, window + size, replica);
+    std::nth_element(replica, replica + (size / 2), replica + size);
+    //std::sort(replica, replica + psize); <--- was used for debug, just leave it as a mark
+
+    PassFilter<int> filter;
+    filter.median = replica[size / 2];
+    filter.upper = filter.median << 3; // median*8
+    filter.lower = filter.median >> 3; // median/8
+
+    return filter;
+}
+
+// This function sums up all values in the array (from p to end),
+// except those that don't fit in the low- and high-pass filter.
+// Returned is the sum and the number of elements taken into account.
+inline std::pair<int, int> AccumulatePassFilter(const int* p, const int* end, PassFilter<int> filter)
+{
+    int count = 0;
+    int sum = 0;
+    for (; p != end; ++p)
+    {
+        // Throw away those that don't fit in the filter
+        if (*p < filter.lower || *p > filter.upper)
+            continue;
+
+        sum += *p;
+        ++count;
+    }
+
+    return std::make_pair(sum, count);
+}
+
+// This function sums up all values in the array (from p to end)
+// and simultaneously elements from `para`, stated it points to
+// an array of the same size. The first array is used as a driver
+// for which elements to include and which to skip, and this is done
+// for both arrays at particular index position. Returner is the sum
+// of the elements passed from the first array and from the `para`
+// array, as well as the number of included elements.
+template <class IntCount, class IntParaCount>
+inline void AccumulatePassFilterParallel(const int* p, const int* end, PassFilter<int> filter,
+        const int* para,
+        int& w_sum, IntCount& w_count, IntParaCount& w_paracount)
+{
+    IntCount count = 0;
+    int sum = 0;
+    IntParaCount parasum = 0;
+    for (; p != end; ++p, ++para)
+    {
+        // Throw away those that don't fit in the filter
+        if (*p < filter.lower || *p > filter.upper)
```

Comment:
To be identical with the previous version should be `<=` and `>=` here.

Reply:
Right! Forgot that De Morgan's laws apply also to relative operators.